### PR TITLE
Remove the use of multiprocessing

### PR DESCRIPTION
**Pull request by kjun9**
This sets `use_multiprocessing=False` in demos in order to avoid tensorflow warnings about using multiprocessing with generators.

See #1006 #1073

### CODE REVIEW

File: demos/link-prediction/hinsage/movielens-recommender.ipynb [hunk 1]
```diff
@@ -469,17 +469,18 @@
...
+      "\tloss: 11.8834\n",
+      "\troot_mean_square_error: 3.4467\n",
+      "\tmean_absolute_error: 3.2562\n"
      ]
     }
    ],
    "source": [
     "test_metrics = model.evaluate_generator(\n",
-    "    test_gen, verbose=1, use_multiprocessing=True, workers=num_workers\n",
+    "    test_gen, verbose=1, use_multiprocessing=False, workers=num_workers\n",
```
Comment (kieranricardo) [inline demos/link-prediction/hinsage/movielens-recommender.ipynb]: We can get rid of the `workers` param here 'cause we're not using multiprocessing.

```suggestion
    "    test_gen, verbose=1, use_multiprocessing=False\n",
```

Comment (kjun9) [reply demos/link-prediction/hinsage/movielens-recommender.ipynb]: I think the `workers` param should get used to do multithreading instead - see https://github.com/stellargraph/stellargraph/pull/925#discussion_r383086045

**Comment by kjun9** [comment]
> I think this doesn't quite address #1006 which is more a performance investigation/seeing whether we should ignore this warning or look at replacing our Sequence objects with tf.data.Datasets. Could you make another issue and connect it instead? I don't want this to get lost!

Yep makes sense 👍 I'll make another issue, but I think we decided to resolve the warnings like this for 1.0, so we might want to remove #1006 from 1.0 then (cc @timpitman )

File: demos/link-prediction/hinsage/movielens-recommender.ipynb (hunk 1 above)
Comment (kieranricardo) [reply demos/link-prediction/hinsage/movielens-recommender.ipynb]: oh good point, missed that in the docs 👍

**Pull request merged**